File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.java
class Solution {
    public int solution(int num) {
        if (num == 1) return 0;
        int count = 0;
        
        while(num != 1) {
            count++;
            if (count >= 500) return -1;

            if (num % 2 == 0) {
                num /= 2;
            } else {
                long result = (long) num * 3 + 1;
                if (result > Integer.MAX_VALUE) return -1;
                num = (int) result;
            }
        }

        return count;
    }
}

File: Kotlin/프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.kt
class Solution {
    private var count = 0
    
    fun solution(a: Int): Int {
        var num: Long = a.toLong()
        while (num != 1L) {
            if (count == 500) return -1
            
            num = getNum(num)
        }
        
        return count
    }
    
    fun getNum(num: Long): Long {
        count++
        return if (num % 2 == 0L) num / 2 else (num * 3) + 1
    }
}